### PR TITLE
REF: Notifications Token handling

### PR DESCRIPTION
**Pull request by marcosrdz**
## Release Notes

- **New Features**
  - Added support for Armenian Dram (AMD) and Serbian Dinar (RSD) as new fiat currencies.

- **Enhancements**
  - Improved notification management, including better handling of permissions and token invalidation.
  - Streamlined notification settings and user experience regarding permission requests.
  - Simplified clipboard functionality in the ReceiveDetails component.

- **Bug Fixes**
  - Enhanced error handling for notification permissions and address fetching processes.
  - Removed options for deferring notification prompts, ensuring a more direct user experience.

### CODE REVIEW

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

File: blue_modules/notifications.js [hunk 1]
```diff
@@ -27,13 +26,42 @@ export const checkNotificationPermissionStatus = async () => {
...
+ * Invalidate the stored push token, removing it from local storage and optionally informing the server.
+ */
+
+export const invalidateToken = async () => {
+  try {
+    const token = await AsyncStorage.getItem(PUSH_TOKEN);
+    if (token) {
+      const parsedToken = JSON.parse(token);
+      if (parsedToken && parsedToken.token) {
+        await fetch(`${baseURI}/invalidateToken`, {
```
Comment (Overtorment) [inline blue_modules/notifications.js:39]: hijo de puta, que es `/invalidateToken`, pendejo?
tu no puedo commit LLM code siempre

**Review by Overtorment — CHANGES_REQUESTED** [review]
notifications actually started working for me.
alli had to do was to go to RECEIVE screen and then app threw a dialog and requested permissions (for some reason permission dialogue was thrown twice).

do we even need this PR anymore? tbh this pr looks like a bunch of scattered code.
if only there were code comments to help me understad why this or that chunk of code was written in the first place...

**Comment by marcosrdz** [comment]
> RECEIVE

yes we should. it allows generating new token and handles "blocked" Status

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Review by Overtorment — CHANGES_REQUESTED** [review]
just tested build *38109 and if you keepturning notifications in the app off an on, it keeps issuing a new token every time.

token should stay the same during the lifetime of an app (with the exception when user forcefully turns off notificaions for BW in OS settings, in that case GC server will  get an error trying to push tosuch tomen and will clenanup this token and all subscriptions from the database; btw this also happens when app is deleted)

**Comment by marcosrdz** [comment]
> just tested build *38109 and if you keepturning notifications in the app off an on, it keeps issuing a new token every time.
> 
> 
> 
> token should stay the same during the lifetime of an app (with the exception when user forcefully turns off notificaions for BW in OS settings, in that case GC server will  get an error trying to push tosuch tomen and will clenanup this token and all subscriptions from the database; btw this also happens when app is deleted) 
> 
> 

That means the user will still get notifications be it silently if they still have a valid token even though the toggled it off. That would defeat the purpose

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by Overtorment** [comment]
waiting for a fresh testflight build to test it

**Review by Overtorment — CHANGES_REQUESTED** [review]
just tested *6447. it still acquires new token every time you disable and enable the pushes

e2e failed as well btw

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Review by Overtorment — APPROVED** [review]
lgtm

**Comment by GladosBlueWallet** [comment]
Unbelievable. You, [subject name here], must be the pride of [subject hometown here]!

**Pull request merged**